galsim/compound.py

```diff
@@ -902,3 +905,252 @@ def __setstate__(self, d):
+        return self._gaussians
+
+    @property
+    def points(self):
+        """
+        getter for the array of points, shape [npoints, 2]
+        """
+        return self._points.copy()
+
+    def _get_gaussians(self, points):
+        """
+        Create galsim.Gaussian objects for each point.
+
+        Highly optimized
+        """
+
+	gaussians = []
+        sigma=self._gaussian_sigma
+        gsparams=self._input_gsparams
+        fluxper=self._flux/self._npoints
+        
+        for p in points:
+            g = galsim._galsim.SBGaussian(
+                sigma=sigma,
+                flux=fluxper,
+                gsparams=gsparams,
+            )
+
+            pos = galsim.PositionD(p[0],p[1])
+
+            g = galsim._galsim.SBTransform(
+                g,
+                1.0,
+                0.0,
+                0.0,
+                1.0,
+                pos,
+                1.0,
+                gsparams,
+            )
+
+            gaussians.append(g)
+
+        return gaussians
+
+    def _set_gaussian_rng(self):
+        """
+        Set the random number generator used to create the points
+
+        We are approximating the random walk to have infinite number
+        of steps, which is just a gaussian
+        """
+
+        # gaussian step size in each dimension for a random walk with infinite
+        # number steps
+        self._sigma_step = self._half_light_radius/2.3548200450309493*2
+
+        self._gauss_rng = galsim.GaussianNoise(
+            self._rng,
+            sigma=self._sigma_step,
+        )
+
+
+    def _get_points(self):
+        """
+        We must use a galsim random number generator, in order for
+        this profile to be used in the configuration file context.
+
+        The most efficient way is to write into an image
+        """
+        ny=self._npoints
+        nx=2
+        im=galsim.ImageD(nx, ny)
+
+        im.addNoise(self._gauss_rng)
+
+        return im.array
+
+    def _verify(self):
+        """
+        type and range checking on the inputs
+        """
+        if not isinstance(self._rng, galsim.BaseDeviate):
+            raise TypeError("rng must be an instance of galsim.BaseDeviate, "
+                            "got %s" % str(self._rng))
+
+        if self._npoints <= 0:
+            raise ValueError("npoints must be > 0, got %s" % str(self._npoints))
+
+        if self._half_light_radius <= 0.0:
+            raise ValueError("half light radius must be > 0"
+                             ", got %s" % str(self._half_light_radius))
+        if self._flux < 0.0:
+            raise ValueError("flux must be >= 0, got %s" % str(self._flux))
+
+    def __str__(self):
+        rep='galsim.RandomWalk(%(npoints)d, %(hlr).16g, flux=%(flux).16g)'
+        rep = rep % dict(
+            npoints=self._npoints,
+            hlr=self._half_light_radius,
+            flux=self._flux,
+        )
+        return rep
+
+    def __repr__(self):
+        return self.__str__()
```

Comment:
The repr (but not the str) should include `gsparams` to be fully descriptive.

Also, the str probably doesn't need 16 digits of precision.  We usually use %s for everything in the str and %r for things in the repr.  For POD, that guarantees that the repr gets all the relevant digits of accuracy, and the str has some reasonable, but not excessive, number of digits.